RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate redux module permissions to typescript

## Proposed changes

Migrate redux module permissions to typescript

## Issue(s)	


## How to test or reproduce

Enter the app and check if the permissions are correct.
Edit a permission and see if it is correct.

## Screenshots

## Types of changes




## Checklist



## Further comments
